toolchain/check/deduce.cpp

```diff
@@ -496,8 +498,46 @@ auto DeductionContext::CheckDeductionIsComplete() -> bool {
       }
       return false;
     }
-  }
 
+    // If the binding is symbolic it can refer to other earlier bindings in the
+    // same generic, or from an enclosing specific. Substitute to replace those
+    // and get a non-symbolic type in order for us to know the final type that
+    // the argument needs to be converted to.
+    //
+    // Note that when typechecking a function declaration, the arguments can
```

Comment:
```suggestion
    // Note that when typechecking a checked generic, the arguments can
```
I think this can happen anywhere where we're performing deduction with a checked generic parameter in scope, eg:

```
class A(T:! type, N:! T) {}
// Deduces T = U in call to A.
class B[U:! type, M:! U](X:! A(U, M)) {}
```